### PR TITLE
PEM 9.5 - Release branch

### PR DESCRIPTION
**Pull request by nidhibhammar**
## What Changed?

PEM 9.5 release tentatively scheduled for 9th Feb

- [PEM-3161](https://enterprisedb.atlassian.net/browse/PEM-3161)
- [PEM-4703](https://enterprisedb.atlassian.net/browse/PEM-4703)
- [PEM-5015](https://enterprisedb.atlassian.net/browse/PEM-5015)

[PEM-3161]: https://enterprisedb.atlassian.net/browse/PEM-3161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PEM-4703]: https://enterprisedb.atlassian.net/browse/PEM-4703?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PEM-5015]: https://enterprisedb.atlassian.net/browse/PEM-5015?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by edbshubham — APPROVED** [review]
LGTM

File: product_docs/docs/pem/9/installing/configuring_the_pem_server_on_linux.mdx [hunk 1]
```diff
@@ -22,24 +22,25 @@ The PEM server package includes a script (`configure-pem-server.sh`) to help aut
...
+| `-uac` or `--use-agent-sslcert`          | Reuses the existing agent SSL certificate while configuring the PEM server.                                                                                                                                                                                                         |
+| `-uak` or `--use-agent-sslkey`           | Reuses the existing agent SSL key while configuring the PEM server.                                                                                                                                                                                                                 |
+| `-scs` or `--server-certificate-subject` | Provides the custom web server certificate subject. The default value includes the EDB details. The format is `/C=CountryCode/ST=StateName/L=CityName/O=CompanySelectionName/CN=CommonNameorHostname/emailAddress=user@company.com`. Provide the `C=CountryCode` as `Alpha-2` code. |
```
Comment (edbshubham) [inline product_docs/docs/pem/9/installing/configuring_the_pem_server_on_linux.mdx:42]: We are not providing any default value, on fresh installation of 9.5 customer has to input the server-certificate-subject else the script will fail

File: product_docs/docs/pem/9/monitoring_event_history.mdx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+| message       | This field provides the description of the action taken for the event. For example, if the user has enabled an alert blackout for the server, it displays `Enabled the alert blackout for the server`.                                                                                                                                                                                                                                                                                                                                                                                |
+| details       | This fields provides the arguments along with its value: <br/><br/>- `AlertBlackoutValue` - displays value `true` if an alert blackout is enabled or disabled.<br/><br/>- `IsAgent` - displays value `true` if alert blackout is for the agent and `false` if alert blackout is for the server.<br/><br/>- `Ids` - diplays the ID of the agent or the server. <br/><br/>- `Scheduled` - displays the value `true` if it is scheduled and `false` if it is configured manually. <br/><br/>- `JobID` - displays the `JobID` if it is scheduled and `null` if it is configured manually. |
```
Comment (edbshubham) [inline product_docs/docs/pem/9/monitoring_event_history.mdx:14]: You can write like:-
This field provides the JSON output for the particular event that happened.
I guess need not to describe the JSON keys inside it to make this doc more generic.
@asheshv What are your thoughts?

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+| Enhancement  | Added `EmailGroup` and `EmailGroupID` in webhook payload.                                                                                                                                                                                                                |
+| Enhancement  | Added a new alert template to validate the password expiry for the users specific to EDB Postgres Advanced Server. Also modified the alert template `A user expired in N days` to `Number of users expiring in N days` that checks the number of users getting exprired. |
+| Enhancement  | Added a button on the Manage Probes window that resets the probe configurations to default.                                                                                                                                                                              |
+| Enhancement  | Added a `Cleared Alert` button in notifications, to mute the cleared alerts. This enhancement provides the flexibility to avoid getting notifications for the cleared alerts.                                                                                            |
```
Comment (sonotley) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:15]:
```suggestion
| Enhancement  | Added a `Cleared Alert` button in notifications. This gives users the flexibility to disable notifications for cleared alerts while still receiving notifications when alerts occur.                                                                                            |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+| Type         | Description                                                                                                                                                                                                                                                              |
+|--------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Enhancement  | Added `EmailGroup` and `EmailGroupID` in webhook payload.                                                                                                                                                                                                                |
+| Enhancement  | Added a new alert template to validate the password expiry for the users specific to EDB Postgres Advanced Server. Also modified the alert template `A user expired in N days` to `Number of users expiring in N days` that checks the number of users getting exprired. |
```
Comment (sonotley) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:13]:
```suggestion
| Enhancement  | Added a new alert template to validate password expiry for users in EDB Postgres Advanced Server. Also modified the alert template `A user expired in N days` to `Number of users expiring in N days` that checks the number of users expiring.

!!! Note
Alerts configured against the outgoing alert template `A user expired in N days` will need to be recreated against the new template.
!!! |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+| Enhancement  | Added a button on the Manage Probes window that resets the probe configurations to default.                                                                                                                                                                              |
+| Enhancement  | Added a `Cleared Alert` button in notifications, to mute the cleared alerts. This enhancement provides the flexibility to avoid getting notifications for the cleared alerts.                                                                                            |
+| Enhancement  | You can now create your own web server certificates as EDB details are no longer added to the web server certificates generated by PEM.                                                                                                                                   |
```
Comment (sonotley) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:16]:
```suggestion
| Enhancement  | You can now provide your own details for web server certificates. EDB details are no longer added to the web server certificates generated by PEM. 

!!! Note
If you are using a certificate generated by PEM, you will be required to provide updated details when updating to PEM 9.5.0 and a new certificate will be generated using these details. This change ensures PEM installations comply with security best practice of including details of the organization responsible for the web server on its certificate.
!!!
|
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+| Enhancement  | Added a capability to identify the user who changed the blackout status, date, and time.                                                                                                                                                                                 |
+| Bug&nbsp;fix | Fixed an issue where the deleted custom probe was still visible in probe dependency tab of Alert Templates.                                                                                                                                                              |
```
Comment (sonotley) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:18]:
```suggestion
| Bug&nbsp;fix | Fixed an issue where a deleted custom probe was still visible in the 'probe dependency' tab of Alert Templates.                                                                                                                                                              |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+| Bug&nbsp;fix | Fixed the missing permission issue by granting the pem_agent user permission to `custom_email_template` table.                                                                                                                                                           |
```
Comment (sonotley) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:19]:
```suggestion
| Bug&nbsp;fix | Fixed a 'missing permission' issue by granting the `pem_agent` user permission to `custom_email_template` table.                                                                                                                                                           |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+| Bug&nbsp;fix | Excluded 0 sized tables from PEM alert template `largest index by table-size percentage`. Also added the detailed SQL statement for this template.                                                                                                                       |
```
Comment (sonotley) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:20]:
```suggestion
| Bug&nbsp;fix | Excluded  zero-sized tables from PEM alert template `largest index by table-size percentage`. Also added a detailed SQL statement for this template.                                                                                                                       |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+| Bug&nbsp;fix | You can now use `v11` as the REST endpoint from the cli.                                                                                                                                                                                                                 |
```
Comment (sonotley) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:21]:
```suggestion
| Bug&nbsp;fix | Fixed an issue whereby the CLI attempted to call an obsolete version of the PEM API, resulting in commands failing.                                                                                                                                                                                                                 |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+| Bug&nbsp;fix | Log in user must have appropriate roles to access `Failover Manager Utilities` and `Database server service management`.                                                                                                                                                 |
```
Comment (sonotley) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:22]: This statement doesn't tell me what we changed. @edbshubham could you clarify and suggest some wording?

**Review by sonotley — CHANGES_REQUESTED** [review]
I've suggested some changes to the release notes. I am not sure if my formatting is valid/good so if necessary we can break the notes out of the table.

File: product_docs/docs/pem/9/installing/configuring_the_pem_server_on_linux.mdx (hunk 1 above)
Comment (nidhibhammar) [reply product_docs/docs/pem/9/installing/configuring_the_pem_server_on_linux.mdx:42]: removed the default value

File: product_docs/docs/pem/9/monitoring_event_history.mdx [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+!!!Note
+At present `pem.event_history` table records the alert blackout history only.
+!!!
+
+| Field name    | Description                                                                                                                                                                                            |
+|---------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| recorded_time | This field records the execution time of the event. It displays the date and time along with the fractional seconds.                                                                                   |
+| user_name     | This field provides the username who executed or scheduled the event.                                                                                                                                  |
+| component     | This field provides the event name.                                                                                                                                                                    |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/monitoring_event_history.mdx:15]:
```suggestion
| component     | The event name.                                                                                                                                                                    |
```

File: product_docs/docs/pem/9/monitoring_event_history.mdx [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+| operation     | This field provides the action taken for the event. For example, if the user has enabled an alert blackout, it displays `enable_alert_blackout`.                                                       |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/monitoring_event_history.mdx:16]:
```suggestion
| operation     | The action taken for the event. For example, if the user has enabled an alert blackout, it displays `enable_alert_blackout`.                                                       |
```

File: product_docs/docs/pem/9/monitoring_event_history.mdx [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+| message       | This field provides the description of the action taken for the event. For example, if the user has enabled an alert blackout for the server, it displays `Enabled the alert blackout for the server`. |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/monitoring_event_history.mdx:17]:
```suggestion
| message       | The description of the action taken for the event. For example, if the user has enabled an alert blackout for the server, it displays `Enabled the alert blackout for the server`. |
```

File: product_docs/docs/pem/9/monitoring_event_history.mdx [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+| details       | This field provides the JSON output of the event that occurred.                                                                                                                                        |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/monitoring_event_history.mdx:18]:
```suggestion
| details       | The JSON output of the event that occurred.                                                                                                                                        |
```

File: product_docs/docs/pem/9/monitoring_performance/alerts.mdx [hunk 15]
```diff
@@ -667,6 +672,8 @@ You can select **Clone** from the top-right corner of the dialog box to clone th
 
 Select **Delete** from the top-right corner of the dialog box to remove a scheduled alert blackout. Select the servers or agents and then select **Delete**.
 
-Select a server for which you want to delete the scheduled alert backout, and then select **Delete**. The server prompts for confirmation before deleting that row.
+Select a server for which you want to delete the scheduled alert blackout, and then select **Delete**. The server prompts for confirmation before deleting that row.
 
 You can select **Reset** to reset the details on the Alert Blackout dialog box to the default settings. Saved blackouts aren't affected.
+
+You can view the scheduled alert blackout details from `event_history` table in `pem` schema once the schedule is executed. For more information, see [monitoring event history](../monitoring_event_history.mdx).
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/monitoring_performance/alerts.mdx:679]:
```suggestion
You can view the scheduled alert blackout details from the `event_history` table in the `pem` schema once the schedule is executed. For more information, see [Monitoring event history](../monitoring_event_history.mdx).
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+---
+
+Released: 09 Feb 2024
+
+New features, enhancements, bug fixes, and other changes in PEM 9.5.0 include:
+
+| Type         | Description                                                                                                                                                                                                                                         |
+|--------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Enhancement  | Added `EmailGroup` and `EmailGroupID` in webhook payload.                                                                                                                                                                                           |
+| Enhancement  | Added a new alert template to validate password expiry for users in EDB Postgres Advanced Server. Also modified the alert template `A user expired in N days` to `Number of users expiring in N days` that checks the number of users expiring.[^*] |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:13]:
```suggestion
| Enhancement  | Added an alert template to validate password expiry for users in EDB Postgres Advanced Server. Also modified the alert template `A user expired in N days` to `Number of users expiring in N days` that checks the number of users expiring.[^*] |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+| Enhancement  | Added a button on the Manage Probes window that resets the probe configurations to default.                                                                                                                                                         |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:14]:
```suggestion
| Enhancement  | Added a button on the Manage Probes window that resets the probe configurations to the defaults.                                                                                                                                                         |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+| Enhancement  | Added a `Cleared Alert` button in notifications. This gives users the flexibility to disable notifications for cleared alerts while still receiving notifications when alerts occur.                                                                |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:15]:
```suggestion
| Enhancement  | Added a **Cleared Alert** button in notifications. This gives you the flexibility to disable notifications for cleared alerts while still receiving notifications when alerts occur.                                                                |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+| Enhancement  | You can now create your own web server certificates as EDB details are no longer added to the web server certificates generated by PEM.                                                                                                             |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:16]:
```suggestion
| Enhancement  | You can now create your own web server certificates, as EDB details are no longer added to the web server certificates generated by PEM.                                                                                                             |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+| Enhancement  | Added a capability to identify the user who changed the blackout status, date, and time.                                                                                                                                                            |
+| Bug&nbsp;fix | Fixed an issue where the deleted custom probe was still visible in probe dependency tab of Alert Templates.                                                                                                                                         |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:18]:
```suggestion
| Bug&nbsp;fix | Fixed an issue where the deleted custom probe was still visible in the **Probe Dependency** tab of Alert Templates.                                                                                                                                         |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+| Bug&nbsp;fix | Fixed the missing permission issue by granting the pem_agent user permission to `custom_email_template` table.                                                                                                                                      |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:19]:
```suggestion
| Bug&nbsp;fix | Fixed the missing permission issue by granting the pem_agent user permission to the `custom_email_template` table.                                                                                                                                      |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+| Bug&nbsp;fix | Log in user must have appropriate roles to access `Failover Manager Utilities` and `Database server service management`.                                                                                                                            |
+| Bug&nbsp;fix | Added `application_name` column in the `session_info` probe.                                                                                                                                                                                        |
+| Bug&nbsp;fix | Fixed the `too few arguments` exception for the webhooks.                                                                                                                                                                                           |
+| Bug&nbsp;fix | Resolved the issue where the `User` column displayed the current username logged into the PEM GUI in the chart display of pg_hba.conf file. Now it displays the respective usernames in the pg_hba.conf file.                                       |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:25]:
```suggestion
| Bug&nbsp;fix | Resolved the issue where the User column displayed the current username logged into the PEM user interface in the chart display of the `pg_hba.conf` file. Now it displays the respective usernames in the `pg_hba.conf` file.                                       |
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx [hunk 23]
```diff
@@ -0,0 +1,29 @@
...
+| Bug&nbsp;fix | Fixed an `Dashboard info not found` error while opening the dashboards in GUI.                                                                                                                                                                      |
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx:26]:
```suggestion
| Bug&nbsp;fix | Fixed a `Dashboard info not found` error while opening the dashboards in the user interface.                                                                                                                                                                      |
```

File: product_docs/docs/pem/9/pem_rest_api.mdx [hunk 24]
```diff
@@ -0,0 +1,5 @@
+---
+title: "PEM Rest API"
+---
+
+EDB Postgres Enterprise Manager supports REST API feature. The REST API feature offers effective data read and write capabilities and access to various functionalities without logging in to the PEM UI. For more information, see [REST API usage](https://www.enterprisedb.com/blog/rest-api-usage-in-pem).
```
Comment (ebgitelman) [inline product_docs/docs/pem/9/pem_rest_api.mdx:5]:
```suggestion
EDB Postgres Enterprise Manager supports the REST API feature. The REST API feature offers effective data read and write capabilities and access to various functionalities without logging in to the PEM UI. For more information, see [REST API usage](https://www.enterprisedb.com/blog/rest-api-usage-in-pem).
```

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx (hunk 10 above)
Comment (sonotley) [reply product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx]: The above comment pertains to line 22.

**Comment by sonotley** [comment]
@nidhibhammar are you able to address my suggestions and questions? There have been many commits subsequently but they don't seem to have addressed the suggestions.

**Comment by nidhibhammar** [comment]
@sonotley, Changes are made as per your suggestions. Only one comment is open now, I will get that fixed after checking with Shubham.

**Review by sonotley — DISMISSED** [review]
LGTM pending that last comment

File: product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx (hunk 10 above)
Comment (edbshubham) [reply product_docs/docs/pem/9/pem_rel_notes/950_rel_notes.mdx]: The user should be a pem_user and should be a member of pem_manage_efm role to access EFM features.
And to start/stop the database service from PEM GUI user should be member of pem_server_service_manager role.

**Review by sonotley — APPROVED** [review]

**Pull request merged**